core/trino-spi/src/main/java/io/trino/spi/type/ShortDecimalType.java

```diff
@@ -42,7 +43,23 @@
 {
     private static final TypeOperatorDeclaration TYPE_OPERATOR_DECLARATION = extractOperatorDeclaration(ShortDecimalType.class, lookup(), long.class);
 
-    ShortDecimalType(int precision, int scale)
+    private static final ShortDecimalType[][] shortDecimalTypes;
```

Comment:
Capitalize SHORT_DECIMAL_TYPES. I might just call it CACHE or INSTANCES, though.